reconcile/openshift_serviceaccount_tokens.py

```diff
@@ -52,13 +53,25 @@ def fetch_desired_state(namespaces, ri, oc_map):
                     ri.register_error()
                 logging.log(level=oc.log_level, msg=oc.message)
                 continue
-            sa_token = oc.sa_get_token(sa_namespace_name, sa_name)
+
+            all_tokens = oc.get_items(kind="Secret", namespace="yihuang")
+            sa_token_list = list(
+                filter(
+                    lambda token: sa_name
+                    == token["metadata"]["annotations"][
+                        "kubernetes.io/service-account.name"
+                    ]
+                    and sa_name + "-token" in token["metadata"]["name"],
+                    all_tokens,
+                )
+            )
+            if sa_token_list:
+                sa_token = sa_token_list[0]["data"]["token"]
```

Comment:
I was just wondering if sa_token_list is sorted in certain order so it can make sure to return the same token.

Reply:
That's a good point, I'll add the sorting. My thought was that {sa_name}-token being present was unique enough to pick out the default token, so theoretically this list should always only have one item, but weird thing can happen.